tools/js-sdk-release-tools/src/changelog/v2/ChangelogGenerator.ts

```diff
@@ -0,0 +1,690 @@
+  private modelPropertyRemovedTemplate = 'Interface {interfaceName} no longer has parameter {propertyName}';
+  private modelPropertyOptionalToRequiredTemplate =
+    'Parameter {propertyName} of interface {interfaceName} is now required';
+  private modelPropertyRequiredToOptionalTemplate =
+    'Parameter {propertyName} of interface {interfaceName} is now optional';
+
+  /** class */
+  // TODO: add more detection for class changes
+  private classAddedTemplate = 'Added Class {className}';
+  private classRemovedTemplate = 'Deleted Class {className}';
+  private classPropertyRemovedTemplate = 'Class {className} no longer has parameter {propertyName}';
+  private classPropertyOptionalToRequiredTemplate = 'Parameter {propertyName} of class {className} is now required';
+  // NOTE: not detected in v1 except constructor and it's parameters
+  private classChangedTemplate = 'Class {className} has a new signature';
+  // NOTE: not in v1
+  private classConstructorAddedTemplate = 'Class {className} has a new constructor "{constructorSignature}"';
+
+  /** type alias */
+  // TODO: add more detection for class changes
+  private typeAliasAddedTemplate = 'Added Type Alias {typeName}';
+  // NOTE: not in v1
+  private typeAliasRemovedTemplate = 'Removed Type Alias {typeName}';
+  private typeAliasTypeChangedTemplate = 'Type alias "{typeName}" has been changed';
+
+  /** function */
+  private functionAddedTemplate = 'Added function {functionName}';
+  private functionRemovedTemplate = 'Removed function {functionName}';
+  private functionAddedOverloadTemplate = 'Added function overload "{functionSignature}"';
+  private functionRemovedOverloadTemplate = 'Removed function overload "{functionSignature}"';
+  private functionSignatureChangedTemplate = 'Function {functionName} has a new signature';
+
+  // TODO: detect enum member's initializer change
+  /** enum */
+  private enumAddedTemplate = 'Added Enum {enumName}';
+  private enumRemovedTemplate = 'Removed Enum {enumName}';
+  private enumMemberAddedTemplate = 'Enum {enumName} has a new value {valueName}';
+  private enumMemberRemovedTemplate = 'Enum {enumName} no longer has value {valueName}';
+
+  private changelogItems: ChangelogItems = {
+    features: new Map<ChangelogItemCategory, string[]>(),
+    breakingChanges: new Map<ChangelogItemCategory, string[]>(),
+  };
+
+  constructor(
+    private detectContext: DetectContext,
+    private detectResult: DetectResult
+  ) {}
+
+  // TODO: add enum support
+  public generate(): ChangelogResult {
+    this.detectResult.interfaces.forEach((diffPairs, name) => {
+      this.generateForInterfaces(diffPairs, name);
+    });
+    this.detectResult.classes.forEach((diffPairs, name) => {
+      this.generateForClasses(diffPairs, name);
+    });
+    this.detectResult.typeAliases.forEach((diffPairs, name) => {
+      this.generateForTypeAliases(diffPairs, name);
+    });
+    this.detectResult.functions.forEach((diffPairs, name) => {
```

Comment:
Is there a ut for default export function name change?

Reply:
do you mean this [test](https://github.com/Azure/azure-sdk-tools/pull/10976/files#diff-7eb7f7930e7c3ad4d28e3e124ce384978fc98e153c830098f6ffe204a2045e35)  in line 81 and 88?